tests/components/blink/test_config_flow.py

```diff
@@ -0,0 +1,204 @@
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mock_blink = Mock(
+        get_auth_token=Mock(return_value=True),
+        key_required=True,
+        login_response={},
+        login_handler=Mock(send_auth_key=Mock(return_value=True)),
+    )
+
+    with patch(
+        "homeassistant.components.blink.config_flow.Blink", return_value=mock_blink
+    ), patch(
+        "homeassistant.components.blink.async_setup", return_value=True
+    ) as mock_setup:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"username": "blink@example.com", "password": "example"}
+        )
+
+    assert result2["type"] == "form"
+    assert result2["step_id"] == "2fa"
+
+    mock_blink.key_required = False
+    with patch(
+        "homeassistant.components.blink.config_flow.Blink", return_value=mock_blink
+    ), patch(
+        "homeassistant.components.blink.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.blink.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"], {"pin": "1234"}
+        )
+
+    assert result3["type"] == "create_entry"
+    assert result3["title"] == "blink"
+    assert result3["result"].unique_id == "blink@example.com"
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass):
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.blink.config_flow.Blink.get_auth_token",
+        return_value=None,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"username": "blink@example.com", "password": "example"}
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_unknown_error(hass):
+    """Test we handle unknown error at startup."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.blink.config_flow.Blink.get_auth_token",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.blink.config_flow.validate_input",
+        side_effect=KeyError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"username": "blink@example.com", "password": "example"}
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "unknown"}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5705368)